### PR TITLE
MWPW-116985 Section Grid Utility

### PR DESCRIPTION
**Pull request by Sartxi**
Adds utility in the section block to render a 12 column grid with options for auto grid and 2-5up column width presets. Includes methods for setting grid width overrides in desktop and grid col/row gap.

Note: I made some updates here to the icon block based on design feedback from consonant and ability to render correctly in the grid. Also, waiting for authoring and engineering feedback to build tests.

Resolves: [MWPW-116985](https://jira.corp.adobe.com/browse/MWPW-116985)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/sarchibeque/section-grids
- After: https://standard-section-grids--milo--adobecom.hlx.page/drafts/sarchibeque/section-grids

### CODE REVIEW

File: libs/blocks/section-metadata/section-metadata.js [hunk 1]
```diff
@@ -12,11 +12,38 @@ function handleBackground(div, section) {
...
-function handleStyle(div, section) {
-  const value = div.textContent.toLowerCase();
-  const styles = value.split(', ').map((style) => style.replaceAll(' ', '-'));
+function handleStyle(div, section, customs = []) {
+  const value = div.textContent?.toLowerCase();
+  let styles = value.split(', ').map((style) => style.replaceAll(' ', '-')).filter(i => i !== '');
   if (section) {
-    section.classList.add(...styles);
+    if (customs.length) styles = [...styles, ...customs];
+    section.classList.add(...styles.filter(i => i !== ''));
```
Comment (meganthecoder) [inline libs/blocks/section-metadata/section-metadata.js:20]: Why are you filtering twice? Seems redundant

File: libs/blocks/section-metadata/section-metadata.js [hunk 2]
```diff
@@ -12,11 +12,38 @@ function handleBackground(div, section) {
...
+  }
+}
+
+function handleColumns(value, values) {
+  if (!value.includes('up')) return { columns: values, offset: null, upClass: null };
+  const ups = { 'two': 6, 'three': 4, 'four': 3, 'five': 2, 2: 6, 3: 4, 4: 3, 5: 2 };
+  const up = values.find(i => i.includes('up'));
```
Comment (meganthecoder) [inline libs/blocks/section-metadata/section-metadata.js:27]: Having `value` and `values` is kind of confusing. Could you just include `values`?
```suggestion
function handleColumns(values) {
  const up = values.find(i => i.includes('up'));
  if (!up) return { columns: values, offset: null, upClass: null };
  const ups = { 'two': 6, 'three': 4, 'four': 3, 'five': 2, 2: 6, 3: 4, 4: 3, 5: 2 };
```

File: libs/styles/variables.css [hunk 3]
```diff
@@ -125,4 +127,4 @@
   :root {
     --grid-container-width: 1200px;
   }
-}
+}
```
Comment (meganthecoder) [inline libs/styles/variables.css:130]: Newline
```suggestion
}

```

**Review by meganthecoder — COMMENTED** [review]
Looks good! Just a few small suggestions. I like what you did turning ups into cols to keep everything on the same grid size 👍

**Comment by meganthecoder** [comment]
Also, tests?

File: libs/blocks/section-metadata/section-metadata.js (hunk 1 above)
Comment (Sartxi) [reply libs/blocks/section-metadata/section-metadata.js:20]: good catch, thanks!

**Comment by Sartxi** [comment]
> Also, tests?

@meganthecoder I was waiting to add tests to be sure the conventions were approved from an authoring and engineering perspective. I will be sure to add them ASAP

File: libs/blocks/icon-block/icon-block.css [hunk 4]
```diff
@@ -29,140 +29,135 @@
   min-height: 700px;
 }
 
-.icon-block .foreground.container {
-  padding-top: var(--spacing-xxxl);
+.icon-block .foreground {
```
[the remark this answers is not in the document]
Comment (Sartxi) [reply libs/blocks/icon-block/icon-block.css:32]: @ryanmparrish is currently working on text blocks and will make the proper updates in tandem

File: libs/blocks/section-metadata/section-metadata.js (hunk 2 above)
Comment (Sartxi) [reply libs/blocks/section-metadata/section-metadata.js:27]: Thanks 👍🏻

**Review by meganthecoder — APPROVED** [review]

**Review by auniverseaway — CHANGES_REQUESTED** [review]
@Sartxi thanks for looking into this! There's a lot of good stuff in here.

When we discussed this idea on the call, I was under the impression we'd have a straight forward solve for 2-up, 3-up, 4-up use cases. This looks to be an entire grid system framework... which is not something I think we want right now.

It would be good to pare this down to the pieces we need today.

**Pull request closed without merging**